From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ivan Pekov <ivan@mrivanplays.com>
Date: Sun, 27 Sep 2020 18:01:50 +0300
Subject: [PATCH] Add last tick time API

// TODO - use Gale's version
Original code by YatopiaMC, licensed under MIT
You can find the original code on https://github.com/YatopiaMC/Yatopia

diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index a2f617537292e3bac52d665a6e51b7d4ce4a227e..2bfed60dc0e046d6a78453c6746ec94420bc9675 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2451,6 +2451,16 @@ public final class Bukkit {
         return server.isStopping();
     }
 
+    // Yatopia start
+    @Deprecated
+    public static long getLastTickMs() {
+        return server.getLastTickMs();
+    }
+    @NotNull public static java.time.Duration getLastTickTime() {
+        return server.getLastTickTime();
+    }
+    // Yatopia end
+
     /**
      * Returns the {@link com.destroystokyo.paper.entity.ai.MobGoals} manager
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index b55c609a6c60659b65d65ee54b233f7a1acdd41b..0f6925a5645ebd742b5a561d4093b3b7f49401b9 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2160,6 +2160,26 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
      */
     boolean isStopping();
 
+    // Yatopia start
+    /**
+     * Returns the time the last tick took in milliseconds.
+     *
+     * @return long time value
+     * @deprecated newer method with java's Duration and a possibility to convert it from millis to something else
+     */
+    @Deprecated
+    default long getLastTickMs() {
+        return getLastTickTime().toMillis();
+    }
+    /**
+     * Returns the time in {@link java.time.Duration} the last tick took.
+     *
+     * @return duration
+     */
+    @NotNull
+    java.time.Duration getLastTickTime();
+    // Yatopia end
+
     /**
      * Returns the {@link com.destroystokyo.paper.entity.ai.MobGoals} manager
      *
